src/core/internal/elf/io.d

```diff
@@ -120,19 +123,13 @@ template ElfIO(Elf_Ehdr, Elf_Shdr, ubyte ELFCLASS)
             enum ELFMAG2 = 'L';
             enum ELFMAG3 = 'F';
 
-            enum ELFCLASS32 = 1;
-            enum ELFCLASS64 = 2;
-
             enum ELFDATA2LSB = 1;
             enum ELFDATA2MSB = 2;
 
             version (LittleEndian)   alias ELFDATA = ELFDATA2LSB;
             else version (BigEndian) alias ELFDATA = ELFDATA2MSB;
             else static assert(0, "unsupported byte order");
 
-            if (fd == -1)
```

Comment:
I don't see any point in moving this `if` at the start of the function. PR's should be minimal in their changes.